src/lib/storage/devicegraph_query.rb

```diff
@@ -0,0 +1,64 @@
+require "storage"
+require "storage/disk_size"
+
+module Yast
+  module Storage
+    # THIS (together with the specs) IS THE MAIN CLASS TO DISCUSS IN THE PULL
+    # REQUEST
+    #
+    # We had quite some logic that was initially in SpaceMaker and then moved to
+    # a ProposalDevicegraph refinement for checking things like
+    #  - available space in the candidate disks (discarding small chunks)
+    #  - partitions in the candidate disks (either directly or with part. table)
+    #  - ...
+    #
+    # While writing the tests, I found myself needing to do similar checks, but
+    # not necessarily restricted to the candidate disks
+    #
+    # This class (together with the Disk refinement it relies on) should allow
+    # to do those "high level" queries, setting a filter (only disk_names for
+    # the time being) in advance to set the scope
+    #
+    # The class is now used in the RSpec tests and in the proposal code
+    class DevicegraphQuery
+      using Refinements::Disk
+
+      # Free disk space below this size will be disregarded
+      TINY_FREE_CHUNK = DiskSize.MiB(30)
+
+      attr_reader :devicegraph
+
+      def initialize(devicegraph, disk_names: nil)
```

Comment:
I was about to suggest to add a block parameter for filtering for most of the query methods, but that should already be covered by this disk_names parameter. OTOH we might still consider passing a block parameter as an alternative (not sure if it's a good approach to pass it to the constructor, though).